### PR TITLE
List phone num

### PR DESCRIPTION
**Pull request by tayponghee**
Add some Junit tests
Add list by phone number

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -78,6 +78,10 @@ Displays a list of all the students that are moved to archive.
 
 Displays a list of all students who were recently deleted.
 
+### Produce a list of students with their phone numbers
+
+Displays a list of all students with their phone numbers.
+
```
Comment (ParthGandhiNUS) [inline docs/UserGuide.md:84]: Good addition to the UG

File: src/main/java/classify/commands/ListStudentsCommand.java [hunk 2]
```diff
@@ -25,6 +25,11 @@ public class ListStudentsCommand {
             "List of students with total classes attended:";
     private static final String INVALID_CHOICE = "Invalid choice!";
     private static final String TOTAL_CLASSES_ATTENDED = " - Total Classes Attended: ";
+    private static final String LIST_PHONE_NUMBERS = "5";
+    private static final String LIST_OF_STUDENTS_WITH_PHONE_NUMBERS = "List of students with phone numbers:";
+    private static final String PHONE_NUM = " - Phone Number: ";
+    private static final String INVALID_PHONE_NUMBER = " - Invalid Phone Number! Save file may be corrupt.";
```
Comment (ParthGandhiNUS) [inline src/main/java/classify/commands/ListStudentsCommand.java:31]: Appropriate use of magic literals

File: src/main/java/classify/commands/ListStudentsCommand.java [hunk 3]
```diff
@@ -146,4 +157,33 @@ private static void listStudentsWithTotalClasses(ArrayList<Student> masterStuden
...
+    /**
+     * Checks if a phone number is valid.
+     *
+     * @param phoneNumber The phone number to be checked.
+     * @return true if the phone number is valid, false otherwise.
+     */
+    private static boolean isValidPhoneNumber(int phoneNumber) {
+        return (phoneNumber >= 80000000 && phoneNumber <= 99999999);
+    }
+
```
Comment (ParthGandhiNUS) [inline src/main/java/classify/commands/ListStudentsCommand.java:188]: Looks good to me.

**Review by alalal47 — COMMENTED** [review]
added functionality to list looks good, will help user in identifying students with same name but different phone number which would have required a tedious series of view commands in the past. LGTM

File: src/test/java/classify/commands/ListStudentsCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        System.setOut(printStream);
+
+        action.run();
+
+        System.out.flush();
+        System.setOut(originalOut);
+        output.add(byteArrayOutputStream.toString().trim());
+        return output;
+    }
+}
```
Comment (ParthGandhiNUS) [inline src/test/java/classify/commands/ListStudentsCommandTest.java:63]: Looks good to me.

**Pull request merged**